engine/classes/Elgg/Notifications/InstantNotificationEvent.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Elgg\Notifications;
+
+use ElggData;
+use ElggEntity;
+
+/**
+ * Instant notification event
+ *
+ * @since 2.3
+ */
+class InstantNotificationEvent implements NotificationEvent {
+
+	use EventSerialization;
+	
+	const DEFAULT_ACTION_NAME = 'notify_user';
+
+	/* @var string The name of the action/event */
+
+	protected $action;
+
+	/* @var string Action's object */
+	protected $object;
+
+	/* @var ElggEntity User who triggered the event */
+	protected $actor;
+
+	/**
+	 * Create a notification event
+	 *
+	 * @param ElggData   $object The object of the event (ElggEntity)
+	 * @param string     $action The name of the action (default: create)
+	 * @param ElggEntity $actor  The entity that caused the event (default: logged in user)
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct(ElggData $object = null, $action = null, ElggEntity $actor = null) {
+
+		$this->object = $object;
+
+		$this->actor = $actor;
+		if (!isset($actor)) {
+			$this->actor = _elgg_services()->session->getLoggedInUser();
+		}
+
+		$this->action = $action ? : self::DEFAULT_ACTION_NAME;
```

Comment:
Huh, I didn't know you could do that. Even though it's been there since PHP 5.3!

> Since PHP 5.3, it is possible to leave out the middle part of the ternary operator.